command/job_restart.go

```diff
@@ -0,0 +1,1185 @@
+	if c.reschedule && len(tasks) != 0 {
+		return 1, fmt.Errorf("The -reschedule option cannot be used with -task")
+	}
+
+	// Dedup tasks and groups.
+	c.groups = set.From(groups)
+	c.tasks = set.From(tasks)
+
+	return 0, nil
+}
+
+// filterAllocs returns a slice of the allocations that should be restarted.
+func (c *JobRestartCommand) filterAllocs(stubs []AllocationListStubWithJob) []AllocationListStubWithJob {
+	result := []AllocationListStubWithJob{}
+	for _, stub := range stubs {
+		shortAllocID := limit(stub.ID, c.length)
+
+		// Skip allocations that are not running.
+		if !stub.IsRunning() {
+			if c.verbose {
+				c.Ui.Output(c.Colorize().Color(fmt.Sprintf(
+					"[dark_gray]    %s: Skipping allocation %q because desired status is %q and client status is %q[reset]",
+					formatTime(time.Now()),
+					shortAllocID,
+					stub.ClientStatus,
+					stub.DesiredStatus,
+				)))
+			}
+			continue
+		}
+
+		// Skip allocations for groups that were not requested.
+		if c.groups.Size() > 0 {
+			if !c.groups.Contains(stub.TaskGroup) {
+				if c.verbose {
+					c.Ui.Output(c.Colorize().Color(fmt.Sprintf(
+						"[dark_gray]    %s: Skipping allocation %q because it doesn't have any of requested groups[reset]",
+						formatTime(time.Now()),
+						shortAllocID,
+					)))
+				}
+				continue
+			}
+		}
+
+		// Skip allocations that don't have any of the requested tasks.
+		if c.tasks.Size() > 0 {
+			hasTask := false
+			for _, taskName := range c.tasks.Slice() {
+				if stub.HasTask(taskName) {
+					hasTask = true
+					break
+				}
+			}
+
+			if !hasTask {
+				if c.verbose {
+					c.Ui.Output(c.Colorize().Color(fmt.Sprintf(
+						"[dark_gray]    %s: Skipping allocation %q because it doesn't have any of requested tasks[reset]",
+						formatTime(time.Now()),
+						shortAllocID,
+					)))
+				}
+				continue
+			}
+		}
+
+		result = append(result, stub)
+	}
+
+	return result
+}
+
+// shouldRestartMultiregion blocks and waits for the user to confirm if the
+// restart of a multi-region job should proceed. Returns true if the answer is
+// positive.
+func (c *JobRestartCommand) shouldRestartMultiregion() bool {
+	question := fmt.Sprintf(
+		"Are you sure you want to restart multi-region job %q in a single region? [y/N]",
+		c.jobID,
+	)
+
+	return c.askQuestion(
+		question,
+		false,
+		func(answer string) (bool, error) {
+			switch strings.TrimSpace(strings.ToLower(answer)) {
+			case "", "n", "no":
+				return false, nil
+			case "y", "yes":
+				return true, nil
+			default:
+				return false, fmt.Errorf("Invalid answer %q", answer)
+			}
+		})
+}
+
+// shouldProceed blocks and waits for the user to provide a valid input on how
+// to proceed. Returns true if the answer is positive.
+//
+// The flags -batch-wait and -on-error have an 'ask' option. This function
+// handles both to prevent asking the user twice in case they are both set to
+// 'ask' and an error happens.
+func (c *JobRestartCommand) shouldProceed(err error) bool {
```

Comment:
I couldn't find a way to test this function since it requires user input and messing with `os.Stdin` during tests seemed dangerous 😬